tests/unit/test_bam_utils.py

```diff
@@ -106,3 +157,59 @@ def test_bam_header_missing_model(bam_header_without_program_group):
     assert ret == False
 
 
+@pytest.mark.parametrize("bam_header_with_multiple_program_groups", ['mas15teloprimev2'], indirect=True)
+def test_load_models_from_bam_header(bam_header_with_multiple_program_groups):
+    with tempfile.NamedTemporaryFile(delete=True) as f:
+        with pysam.AlignmentFile(f.name, "wb", header=bam_header_with_multiple_program_groups) as bf:
+            bf.close()
+
+        models = bam_utils.load_models(None, input_bam=f)
+
+        assert len(models) == 2
+        assert models[0].name == 'mas15v2'
+        assert models[1].name == 'mas15teloprimev2'
+
+
+def test_load_model_from_name():
+    for model_name in list(model.LibraryModel.pre_configured_models.keys()):
+        lb_models = bam_utils.load_models([model_name])
+
+        assert lb_models[0].name == model_name
+
+
+def test_load_model_from_json():
+    with tempfile.NamedTemporaryFile(delete=False) as f:
+        f.write(b'{"name":"mas3teloprimev2","description":"An example model for a 3-element array.","version":"1.0.0","array_element_structure":[["A","TPV2_adapter","cDNA","Poly_A","idx","rev_bind"],["B","TPV2_adapter","cDNA","Poly_A","idx","rev_bind"], ["C","TPV2_adapter","cDNA","Poly_A","idx","rev_bind","D"]],"adapters":{"TPV2_adapter":"CTACACGACGCTCTTCCGATCTTGGATTGATATGTAATACGACTCACTATAG","rev_bind":"CTCTGCGTTGATACCACTGCTT","A":"AGCTTACTTGTGAAGAT","B":"ACTTGTAAGCTGTCTAT","C":"ACTCTGTCAGGTCCGAT","D":"ACCTCCTCCTCCAGAAT","Poly_A":{"HomopolymerRepeat":["A",30]},"idx":{"FixedLengthRandomBases":10},"cDNA":"random"},"direct_connections":{"A":["TPV2_adapter"],"B":["TPV2_adapter"],"C":["TPV2_adapter"],"D":["TPV2_adapter"],"TPV2_adapter":["cDNA"],"cDNA":["Poly_A"],"Poly_A":["idx"],"idx":["rev_bind"],"rev_bind":["B","C","D"]},"start_element_names":["A","TPV2_adapter"],"end_element_names":["rev_bind","D"],"named_random_segments":["idx","cDNA"],"coding_region":"cDNA","annotation_segments":{"idx":[["BC","XB"]]}}\n')
+
+    lb_models = bam_utils.load_models([f.name])
+
+    assert lb_models[0].name == "mas3teloprimev2"
+
+
+def test_reverse_complement():
+    d = {
+        'GTTCTAGCGCTAGTATG': 'CATACTAGCGCTAGAAC',
```

Comment:
Can you add a couple of tests here for the IUPAC base reverse complementing?

Reply:
Fixed.